api/server/server.go

```diff
@@ -138,8 +138,13 @@ func (s *Server) makeHTTPHandler(handler httputils.APIFunc) http.HandlerFunc {
 		}
 
 		if err := handlerFunc(ctx, w, r, vars); err != nil {
-			logrus.Errorf("Handler for %s %s returned error: %+v", r.Method, r.URL.Path, err)
-			httputils.MakeErrorHandler(err)(w, r)
+			statusCode := httputils.GetHTTPErrorStatusCode(err)
+			errFormat := "%v"
+			if statusCode == http.StatusInternalServerError {
+				errFormat = "%+v"
+			}
+			logrus.Errorf("Handler for %s %s returned error: "+errFormat, r.Method, r.URL.Path, err)
+			httputils.MakeErrorHandler(err, statusCode)(w, r)
```

Comment:
@aaronlehmann @vdemeester I changed MakeErrorHandler to be able to pass a statusCode, so that GetHTTPErrorStatusCode doesn't get called twice. @thaJeztah thinks it's unnecessary, I could agree, what do you think? Should I make the change?